vNext/src/Microsoft.AspNet.OData/EdmLibHelpers.cs

```diff
@@ -188,7 +188,20 @@ public static IEdmTypeReference GetEdmTypeReference(this IEdmModel edmModel, Typ
                 Type elementType;
                 if (clrType.IsCollection(out elementType))
                 {
-                    edmType = new EdmCollectionType(edmModel.GetEdmType(elementType).ToEdmTypeReference(IsNullable(elementType)));
+                    //edmType = new EdmCollectionType(edmModel.GetEdmType(elementType).ToEdmTypeReference(IsNullable(elementType)));
```

Comment:
The line should have been removed. The IEdmTypeReference is discovered first, then the EdmCollectionType is created.